fix test_dispatch tests to error on duplicate def

bdhirsh opened the pull request:
fbcode shows some `test_dispatch.py` tests failing after my previous PR here: https://github.com/pytorch/pytorch/pull/48098

https://www.internalfb.com/tasks/?t=81313176

I fixed some failing tests to expect a duplicate def error, and updated the error strings expected in the tests by running `EXPECTTEST_ACCEPT=1 python test/test_dispatch.py`, as indicated in `torch/testing/_internal/expecttest.py`.

Should these tests run in OSS? It looks like they currently don't (it looks like they also didn't run in my phabricator diff).

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49254 fix test_dispatch tests to error on duplicate def**

Differential Revision: [D25505170](https://our.internmc.facebook.com/intern/diff/D25505170)

bdhirsh committed:
fix test_dispatch tests to error on duplicate def

[ghstack-poisoned]

bdhirsh committed:
Update on "fix test_dispatch tests to error on duplicate def"


Differential Revision: [D25505170](https://our.internmc.facebook.com/intern/diff/D25505170)

[ghstack-poisoned]

bdhirsh committed:
Update on "fix test_dispatch tests to error on duplicate def"

fbcode shows some `test_dispatch.py` tests failing after my previous PR here: https://github.com/pytorch/pytorch/pull/48098

https://www.internalfb.com/tasks/?t=81313176

I fixed some failing tests to expect a duplicate def error, and updated the error strings expected in the tests by running `EXPECTTEST_ACCEPT=1 python test/test_dispatch.py`, as indicated in `torch/testing/_internal/expecttest.py`.

Should these tests run in OSS? It looks like they currently don't (it looks like they also didn't run in my phabricator diff).


Differential Revision: [D25505170](https://our.internmc.facebook.com/intern/diff/D25505170)

[ghstack-poisoned]

ezyang commented:
They should be running on OSS. At a guess, test_dispatch isn't listed in `test/run_test.py` and that might be the problem. I suggest patching them in there and seeing if that fixes it.

bdhirsh committed:
Update on "fix test_dispatch tests to error on duplicate def"

fbcode shows some `test_dispatch.py` tests failing after my previous PR here: https://github.com/pytorch/pytorch/pull/48098

https://www.internalfb.com/tasks/?t=81313176

I fixed some failing tests to expect a duplicate def error, and updated the error strings expected in the tests by running `EXPECTTEST_ACCEPT=1 python test/test_dispatch.py`, as indicated in `torch/testing/_internal/expecttest.py`.

Should these tests run in OSS? It looks like they currently don't (it looks like they also didn't run in my phabricator diff).


Differential Revision: [D25505170](https://our.internmc.facebook.com/intern/diff/D25505170)

[ghstack-poisoned]

bdhirsh committed:
Update on "fix test_dispatch tests to error on duplicate def"

fbcode shows some `test_dispatch.py` tests failing after my previous PR here: https://github.com/pytorch/pytorch/pull/48098

https://www.internalfb.com/tasks/?t=81313176

I fixed some failing tests to expect a duplicate def error, and updated the error strings expected in the tests by running `EXPECTTEST_ACCEPT=1 python test/test_dispatch.py`, as indicated in `torch/testing/_internal/expecttest.py`.

Should these tests run in OSS? It looks like they currently don't (it looks like they also didn't run in my phabricator diff).


Differential Revision: [D25505170](https://our.internmc.facebook.com/intern/diff/D25505170)

[ghstack-poisoned]
